From 9bb88ee15c3b603c39b6140c040b4a5b17c8a86a Mon Sep 17 00:00:00 2001
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Fri, 30 Mar 2018 22:25:57 +0200
Subject: [PATCH 1/1] NetworkManager: allow to wait on boot

This patch allow a system to wait for network card on boot time.

Signed-off-by: Pascal Vizeli <pvizeli@syshack.ch>
---
 package/network-manager/Config.in          | 6 ++++++
 package/network-manager/network-manager.mk | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/package/network-manager/Config.in b/package/network-manager/Config.in
index 72658c1..1bdb225 100644
--- a/package/network-manager/Config.in
+++ b/package/network-manager/Config.in
@@ -62,6 +62,12 @@ config BR2_PACKAGE_NETWORK_MANAGER_OVS
 	help
 	  This option enables support for OpenVSwitch
 
+config BR2_PACKAGE_NETWORK_MANAGER_WAIT_BOOT
+	bool "Wait on boot support"
+	depends on BR2_INIT_SYSTEMD
+	help
+	  This option allow NetworkManager to wait on boot
+
 endif
 
 comment "NetworkManager needs udev /dev management and a glibc toolchain w/ headers >= 3.7, dynamic library"
diff --git a/package/network-manager/network-manager.mk b/package/network-manager/network-manager.mk
index a520aad..d48ffe9 100644
--- a/package/network-manager/network-manager.mk
+++ b/package/network-manager/network-manager.mk
@@ -102,6 +102,12 @@ define NETWORK_MANAGER_INSTALL_INIT_SYSTEMD
 
 	ln -sf /usr/lib/systemd/system/NetworkManager-dispatcher.service \
 		$(TARGET_DIR)/etc/systemd/system/dbus-org.freedesktop.nm-dispatcher.service
+
+	if [ $(BR2_PACKAGE_NETWORK_MANAGER_WAIT_BOOT) == y ]; \
+	then \
+		ln -sf /usr/lib/systemd/system/NetworkManager-wait-online.service \
+		$(TARGET_DIR)/etc/systemd/system/multi-user.target.wants/NetworkManager-wait-online.service; \
+	fi
 endef
 
 $(eval $(autotools-package))
-- 
2.7.4

